### PR TITLE
add package nowide

### PR DESCRIPTION
**Pull request by jingkaimori**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/n/nowide_standalone/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package("nowide_standalone")
+
+    set_kind("library")
```
Comment (waruqi) [inline packages/n/nowide_standalone/xmake.lua:3]: you need not add it.

File: packages/n/nowide_standalone/xmake.lua [hunk 2]
```diff
@@ -15,12 +15,5 @@ package("nowide_standalone")
...
     on_test(function (package)
-        assert(package:check_cxxsnippets({test = [[
-            #include <nowide/args.hpp>
-            int test(int argc, char **argv)
-            {
-                nowide::args _(argc, argv); // Must use an instance!
-                return argc;
-            }
-        ]]}))
+        assert(package:has_cxxtypes("nowide::args",{includes={"nowide/args.hpp","windows.h"}}))
```
Comment (waruqi) [inline packages/n/nowide_standalone/xmake.lua:18]: please revert it. cxxsnippet will check links

File: packages/n/nowide_standalone/xmake.lua [hunk 3]
```diff
@@ -11,9 +9,17 @@ package("nowide_standalone")
```
Comment (waruqi) [inline packages/n/nowide_standalone/xmake.lua:9]:
```lua
if is_plat("windows") then
    add_syslinks("shell32")
end
```

File: packages/n/nowide_standalone/xmake.lua [hunk 4]
```diff
@@ -11,9 +9,17 @@ package("nowide_standalone")
 
     add_deps("cmake")
     on_install("windows", "macosx", "linux", function (package)
+        package:add("links","shell32")
```
Comment (waruqi) [inline packages/n/nowide_standalone/xmake.lua:12]: use add_syslinks

**Comment by jingkaimori** [comment]
mingw port could be tested

File: packages/n/nowide_standalone/xmake.lua [hunk 5]
```diff
@@ -8,12 +8,12 @@ package("nowide_standalone")
     add_versions("11.2.0", "1869d176a8af389e4f7416f42bdd15d6a5db3c6e4ae77269ecb071a232304e1d")
 
     add_deps("cmake")
-    
-    if is_plat("windows") then
+
+    if is_plat("windows") or is_plat("mingw") then
```
Comment (waruqi) [inline packages/n/nowide_standalone/xmake.lua:12]: is_plat("windows", "mingw")

**Comment by jingkaimori** [comment]
please merge by squash

**Pull request merged**